pandas/core/arrays/datetimelike.py

```diff
@@ -2519,8 +2519,13 @@ def _validate_inferred_freq(
     -------
     freq : DateOffset or None
     """
+    offset1 = to_offset(freq)
+    offset2 = to_offset(inferred_freq)
+
+    freq_equal = type(offset1) == type(offset2)
```

Comment:
I do not think this is sufficient, for example `QuarterBegin: startingMonth=2` and `QuarterBegin: startingMonth=3` will both have the same type, but are not the same frequency.